.github/workflows/api_reference.yaml

```diff
@@ -0,0 +1,41 @@
+name: api_reference
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    # Only run api_reference action when a header file has been modified.
+    paths:
+      - '**.hpp'
```

Comment:
Maybe it makes more sense if the files are in the include folder because some header files are going to be private and the doxygen documentation should not be included in the API reference in this case.